### PR TITLE
fix same missing dependencies

### PR DESCRIPTION
**Pull request by InuSasha**
Add missing dependencies, so packages build via ./script/install 
So i could find some missing dependencies.

The host-packages (not incl. in toolchain is self) needs a working build environment, so i had easly added toolchain. With more investigation, we can reduce it gcc:host, ccache:host. autotools, etc., but i think this is to many work, without perceptible result.

I only checked packages needed by Generic and RPi2 builds.

Finale Generic & RPi2 build was fine, and all single package builds, too ;)

### CODE REVIEW

**Comment by lrusak** [comment]
Thanks, I'll spin up a build on jenkins

**Comment by stefansaraev** [comment]
meh, because.

1. scripts/install was never supposed to be called manualy. and it should not be used that way to "detect missing dependencies".
2. host packages should never depend on LE's own toolchain.
3. also, gcc:host is not the host's gcc. host packages use your host system's gcc. you dont need gcc:host or anything else from the toolchain

**Comment by InuSasha** [comment]
@stefansaraev 
to 1. is was not designed for it, yes. but i was working. (may script/build will better)
to 2-4. the system build environment is created in toolchain (or a packages it depends on). You has written "you are allowed to use scripts/build once toolchain is built." I think this is the implicit dependency, didn't it? What i make is, define it explicit, now.
Of course, there may be better solution, as mine.

**Comment by stefansaraev** [comment]
no it does not work, as seen in this pr, and it wont work without a bit of rework. look at the begining of scripts/image. there is a bunch of special packages, including toolchain and few :host packages like squashfs, kmod and fakeroot, which are essential for the build system. there are good reasons for that, one of it is avoiding potential circular dependencies :)

there is zero reasons to do scripts/install randompackage:host. it should never be done. there is zero reasons to use scripts/install for a :target package too. scripts/install is supposed to install package's files to target image, before making squashfs root. that's all.

so, in short, if you want to use scripts/build by hand, you should at least:
1. call scripts/checkdeps
2. do scripts/build toolchain

for the build system to build a :host packages, it needs /usr/bin/gcc and few other tools (as said, scripts/checkdeps should take care), and few other :host packages (autotools, libtool and such). if you dont like how it's done, you can make yet-another virtual package with those required packages. adding toolchain as a dependency for any host package for your specific use case is just wrong.

btw, why do you think yasm:host needs target gcc ?

**Comment by InuSasha** [comment]
@stefansaraev 
yasm: ok it needs only ccache:host, it create the required host-gcc in toolchain.

we can define toolchain has to build first, in the script/image (like it is). So we can remove toolchain from every depend-list.

**Comment by stefansaraev** [comment]
$ROOT/$TOOLCHAIN/bin/host-gcc is created in ccache:host. just add ccache:host to yasm:host depends.

in future, I suggest moving ccache:host build to the beginning of scripts/image, before toolchain, and removing it from all pkg depends.

fyi I am in no way against this PR, neither I am in a position to block or approve it. I dont even use libreelec atm. it's up to the people who have access to press the merge button ;)

**Comment by InuSasha** [comment]
changes dependency from yasm to ccache:host (instead of gcc:host) and moved to commit 1.

@stefansaraev i have no problems with your comments :) (only with the/my english XD).
some of the points, i had by meself (that's why it split this pr into 2 commits)
more brains, more ideas

**Comment by InuSasha** [comment]
any objectives?

**Comment by lrusak** [comment]
not 100% sure about this really

**Comment by InuSasha** [comment]
should i split it into 2 PRs?
I think, the first is completely unproblematic.

**Comment by lrusak** [comment]
I think I am ok with this. Has anyone build tested it?

@InuSasha can you rebase?

**Comment by InuSasha** [comment]
i actual build all projects. will comment, when done

**Comment by InuSasha** [comment]
all project was building fine*

(for generic, #1086 was also needed, but this is independed from this PR)

**Comment by InuSasha** [comment]
rebase to master, only
